modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/repository/RepositoryDistributionSnapshot.java

```diff
@@ -434,4 +447,128 @@ public DistributionSnapshot readJson(InputStream in) {
                         .stream()
                         .collect(Collectors.toMap(Plugin::getId, p -> p.getRepositorySnapshot(getDoc())));
     }
+
+    /**
+     * Returns a key/value map of properties for update.
+     *
+     * @since 11.2
+     */
+    public Map<String, String> getUpdateProperties() {
+        Map<String, String> props = new HashMap<>();
+        Stream.of(TITLE_PROPERTY_PATH, PROP_NAME, PROP_VERSION, PROP_KEY).forEach(p -> props.put(p, safeGet(p)));
```

Comment:
For basic loops without filtering/mapping we should prefer `List.of(...).forEach()` to avoid building a stream.
There are several in this method and the following ones.